language-support/ts/daml-types/index.ts

```diff
@@ -696,3 +696,26 @@ export const Map = <K, V>(
   encode: (m: Map<K, V>): unknown =>
     m.entriesArray().map(e => [kd.encode(e[0]), vd.encode(e[1])]),
 });
+
+/**
+ * A disclosed contract that can be passed on a command submission.
+ * @property contractId the contract id of the contract.
+ * @property templateId the template id of the contract.
+ * @property payload The encoded payload of the contract as obtained from the {@link Serializable.encode} method.
+ * @property metadata The contract metadata
+ */
+export type DisclosedContract = {
+  contractId: ContractId<unknown>;
+  templateId: string;
+  payload: unknown;
```

Comment:
Can/should we be more specific than `unknown` here? I'm not very familiar with Typescript, but could it be something like `object | string` or something?

Reply:
This needs to match what the `encode` method gives you. That's an `unknown` atm.

Reply:
Fair enough.